src/api/java/com/minecolonies/api/quests/IQuestReward.java

```diff
@@ -0,0 +1,18 @@
+package com.minecolonies.api.quests;
+
+import com.minecolonies.api.colony.IColony;
+import net.minecraft.world.entity.player.Player;
+
+/**
+ * Quest objective interface for all objective types.
+ */
+public interface IQuestReward
+{
+    /**
+     * Apply the reward to colony and player.
+     * @param colony the involved colony.
+     * @param player the involved player.
+     * @param colonyQuest the related quest.
+     */
+    void applyReward(final IColony colony, final Player player, final IColonyQuest colonyQuest);
+}
```

Comment:
should all rewards maybe get a displaye component they return for displaying them easier?

Reply:
Can be added in the future when we add things to display them